pyensign/connection.py

```diff
@@ -94,8 +107,32 @@ def __init__(
         else:
             self.client_id = client_id
 
+    def _ensure_ready(self):
+        """
+        Create the gRPC channel and stub if not already connected. This is done at call
+        time rather than when the client is created to ensure that user code runs in
+        the same event loop as the gRPC library. This function also creates the worker
+        pool and shutdown event signal if not already created.
+        TODO: We could avoid this by requiring the Ensign client to be created in the
+        event loop, e.g. async with Client(...) as client: ..., but the current syntax
+        is a bit easier.
```

Comment:
What is your opinion on this syntax

```python
async with Ensign() as ensign:
    await ensign.publish(...)

or

ensign = await Ensign()
```

vs. how it's currently done?

```python
ensign = Ensign()
```

So we would be requiring users to already be in a coroutine with the former option, but it does mean that the API is a bit more consistent, because everything would be async.